src/main/java/org/springframework/cloud/deployer/spi/cloudfoundry/CloudFoundryAppDeployer.java

```diff
@@ -99,13 +103,19 @@ public String deploy(AppDeploymentRequest request) {
 
 		Map<String, String> envVariables = new HashMap<>();
 
-		try {
-			envVariables.put("SPRING_APPLICATION_JSON",
-					new ObjectMapper().writeValueAsString(
-							Optional.ofNullable(request.getDefinition().getProperties())
-									.orElse(Collections.emptyMap())));
-		} catch (JsonProcessingException e) {
-			throw new RuntimeException(e);
+		if (useSpringApplicationJson(request)) {
```

Comment:
The usage should be that SAJ needs to be the default, and one needs to opt into using env-vars.  Otherwise we have a bad experience to specify this property all the time to deploy to PCF for the data flow use cases (where we can't rely on env-vars as they are too brittle with relaxed bindings for boot apps)